win32_compat/README

```diff
@@ -0,0 +1,28 @@
+# Windows compatibility layer
+
+This folder contains headeres and source code to make libsolv compile using MSVC
+in a Windows environment. Most of the sources contained in this folder were
+extracted from `musl`, a MIT licensed libc implementation. Below you find a 
+listing of all files, and their origin:
+
+- musl (MIT License)
+  - features
+  - fnmatch
+  - getopt
+  - regerror
+  - regex
+  - strcasecmp/strncasecmp
+  - strcasestr
+
+- stackoverflow (CC-BY-SA)
+  - unistd (https://stackoverflow.com/a/826027/1347553)
+  - pread (https://stackoverflow.com/a/42495892)
```

Comment:
I'm unsure if CC-BY-SA is compatible with all the downstream users' licensing...

Reply:
Hmm. We could remove unistd completely (before I was commenting it out everywhere and I think libsolv will build without it.
For pread we can also take the one from postgres sources and modify it.

Reply:
removed all cc-by-sa